### DISCUSSION TITLE
[Hackathon Use Case] FDC3 Java API

### DISCUSSION THREAD

**Discussion by maoo** [finos/finos-pmcs#256] [Ideas]
# FDC3 Java API
_From Wellington_

## Business Problem
Some firms have existing Java desktop applications, and they want to use FDC3 to integrate with other apps that use JavaScript or other technologies.

For example, a buy-side trader using an internal Java order management system selects an order on their blotter and wants to view related analytics in an external JavaScript app provided by a broker.

OpenFin has some support for FDC3 integration for Java apps using their Interop API. However, this is vendor-specific code that isn’t the same as the JavaScript/TypeScript FDC3 API.

Ideally FINOS should have interfaces for a FDC3 Java API, which can be implemented by OpenFin or other FDC3 desktop agent vendors.

This would provide a standardized API for Java app developers to use, making it easier to switch the underlying technology that provides the FDC3 communication if required.

This is a similar approach to the [FDC3 .NET API](https://github.com/finos/FDC3/issues/931) that was built by Morgan Stanley.

## Expected outcome
- Create a vendor-agnostic open-source FDC3 Java API, that matches the FDC3 TypeScript API as closely as possible.
- Create an OpenFin implementation of this API using the existing OpenFin Java Adapter. This may be a partial implementation if some features aren’t currently supported by OpenFin.
- Demonstrate that we can send FDC3 context or intents from a Java to JavaScript app using this library.

## Useful apps, systems, projects, technologies, libraries and datasets
- Java app (Swing, JavaFX, or other UI technology)
- FDC3 JavaScript app (can use [FDC3 Workbench](https://fdc3.finos.org/toolbox/fdc3-workbench/) or similar example app for POC)
- [OpenFin Java Adapter](https://developers.openfin.co/of-docs/docs/java-api)

**Comment by TheJuanAndOnly99** [comment]
Closing discussion in preparation [for this years hackathon](https://www.finos.org/hosted-events/2024-10-25-citi-india-hackathon-supported-by-finos). Feel free to reopen and update if its still relevant.
